src/wallet/wallet.cpp

```diff
@@ -2006,23 +1978,15 @@ CAmount CWalletTx::GetAvailableCredit(interfaces::Chain::Lock& locked_chain, boo
     if (pwallet == nullptr)
         return 0;
 
+    bool got_cache = filter == ISMINE_SPENDABLE || filter == ISMINE_WATCH_ONLY;
```

Comment:
Is purpose of the got_cache variable just to avoid changing behavior in this PR? If so, would suggest renaming and commenting:

```c++
// Avoid caching ismine or NONE or ALL cases (could remove this check and simplify in the future).
const bool allow_cache = filter == ISMINE_SPENDABLE || filter == ISMINE_WATCH_ONLY;
```

Reply:
Makes sense, changed.